ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py

```diff
@@ -607,4 +607,50 @@ def __init__(self, config: WCA_PIPELINE_CONFIGURATION):
         super().__init__(config=config)
 
     def invoke(self, params: RoleExplanationParameters) -> RoleExplanationResponse:
-        raise FeatureNotAvailable
+        request = params.request
+        files = params.files
+        model_id = params.model_id
+        explanation_id = params.explanation_id
+
+        organization_id = request.user.organization.id if request.user.organization else None
+        api_key = self.get_api_key(request.user, organization_id)
+        model_id = self.get_model_id(request.user, organization_id, model_id)
+
+        headers = self.get_request_headers(api_key, explanation_id)
+        data = {
+            "role_name": params.role_name,
+            "model_id": model_id,
+            "files": files,
+        }
+
+        @backoff.on_exception(
+            backoff.expo,
+            Exception,
+            max_tries=self.retries + 1,
+            giveup=self.fatal_exception,
+            on_backoff=self.on_backoff_explain_playbook,
+        )
+        @wca_explain_playbook_hist.time()
```

Comment:
Probably want a different metric.

Reply:
Thank you @manstis, I'm fixing them both.